backend/src/models/Table.js

```diff
@@ -0,0 +1,79 @@
+const db = require('./db');
+
+function validate(table) {
+  if (table.tableName === undefined) {
+    throw Error(message='DB table name is not set');
+  };
+  if (table.pk === undefined) {
+    throw Error(message='PK name is not set');
+  };
+  if (table[table.pk] === undefined) {
+    throw Error(message=`'${table.pk}' is not set`);
+  }
+}
+
+class Table {
+  constructor(tableName, pk, data) {
+    this.tableName = tableName;
+    this.pk = pk;
+    if (pk in data) {
```

Comment:
Do we need to add pk separately from other fields and directly on this?

Reply:
It make sense to separate abstract `Table` from concrete `User`. Table must have table `name` and `pk` to function with any table so they are in the `Table` constructor.
On the other hand, `data` as the object of arbitrary content passed in one step, it is convenient, you can validate it later.

Reply:
pk is just one of the columns in the table, frankly I don't see how it's different from the other columns. And don't you need to be validating it along with the other properties?

Reply:
You're right, I mixed up `this.pk` with `this.data[this.pk]`.